spec/controllers/user_journey_controller_spec.rb

```diff
@@ -100,27 +101,43 @@
 
   context '#upload_certificate ' do
     it 'renders upload certificate page' do
-      certmgr_stub_auth
+      certmgr_stub_auth(team)
       get :upload_certificate, params: params
       expect(response).to have_http_status(:success)
       expect(subject).to render_template(:upload_certificate)
     end
+
+    it 'does not render the upload certificate page when team is not matching' do
+      certmgr_stub_auth
+      foreign_team = FactoryBot.create(:team, id: SecureRandom.uuid)
+      foreign_component = FactoryBot.create(:sp_component, team_id: foreign_team.id)
```

Comment:
Non blocking; FactoryBot is already loaded into rspec in the background so `create` and `build` an be called directly without referencing `FactoryBot`.